src/cpp/session/SessionModuleContext.cpp

```diff
@@ -1156,14 +1156,27 @@ FilePath findProgram(const std::string& name)
    else
    {
       std::string which;
+      FilePath resultPath;
+
+      // TODO: switch over to using findProgramOnPath all the time here - for now, doing it both ways to ensure there
+      // are no edge cases in findProgramOnPath that make a difference to app code
       Error error = r::exec::RFunction("Sys.which", name).call(&which);
+      Error dbgError = system::findProgramOnPath(name, &resultPath);
```

Comment:
For some extra context, we've seen things like this manifest on Windows, which (as I understand it) is caused by the fact that R links statically to the C runtime, whereas RStudio dynamically links to the C runtime. Because there are two separate C runtimes, there are also two separate `environ` blocks (among other things) which can cause issues if we need to ensure we're respecting the PATH as seen and mutated by the R process.

Some other examples where we've had to do similar gymnastics (mainly just FYI; not directly relevant for this PR):

https://github.com/rstudio/rstudio/blob/30872eaea5fce0fe66ded0f91bfe16cf24dbe0fd/src/cpp/r/include/r/RUtil.hpp#L21-L26

https://github.com/rstudio/rstudio/blob/30872eaea5fce0fe66ded0f91bfe16cf24dbe0fd/src/cpp/r/RUtil.cpp#L323-L350

Reply:
So maybe for windows, we'll need to get hold of R's version of the PATH variable during init. Thanks, that's a super helpful piece of info to keep in mind as well!